src/apps/tools/permissions.clj

```diff
@@ -2,9 +2,24 @@
   (:use [clojure-commons.error-codes :only [clj-http-error?]]
         [slingshot.slingshot :only [try+ throw+]])
   (:require [apps.clients.permissions :as permissions]
+            [apps.persistence.tools :as tools-db]
+            [apps.util.service :as service]
             [clojure-commons.exception-util :as exception-util]
             [clojure.string :as string]))
 
+(defn- list-non-existent-tool-ids
+  [tool-id-set]
+  (->> (tools-db/get-tools-by-id tool-id-set)
+       (map :id)
+       (set)
+       (clojure.set/difference tool-id-set)))
```

Comment:
This was just a copy of the [jobs version of this code](https://github.com/cyverse-de/apps/blob/master/src/apps/persistence/jobs.clj#L593-L598) 😁